File: festival-web/src/main/java/cn/bdqfork/web/route/ModelAndView.java
package cn.bdqfork.web.route;

import java.util.HashMap;
import java.util.Map;

/**
 * @author bdq
 * @since 2020/2/21
 */
public class ModelAndView {
    private String view;
    private Map<String, Object> model;

    public ModelAndView(String view) {
        this(view, new HashMap<>());
    }

    public ModelAndView(String view, Map<String, Object> model) {
        this.view = view;
        this.model = model;
    }

    public void add(String key, Object value) {
        model.put(key, value);
    }

    public String getView() {
        return view;
    }

    public Map<String, Object> getModel() {
        return model;
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/ModelAndView.kt
package cn.bdqfork.kotlin.web.route

import java.util.*

/**
 * @author bdq
 * @since 2020/2/21
 */
class ModelAndView @JvmOverloads constructor(val view: String, private val model: MutableMap<String, Any?> = HashMap()) {
    fun add(key: String, value: Any?) {
        model[key] = value
    }

    fun getModel(): Map<String, Any?> {
        return model
    }

}